perun-core/src/main/java/cz/metacentrum/perun/core/entry/ServicesManagerEntry.java

```diff
@@ -60,6 +63,227 @@ public ServicesManagerEntry(PerunBl perunBl) {
+		}
+		return getServicesManagerBl().getServicesBlockedOnFacility(perunSession, facility);
+	}
+
+	@Override
+	public List<Service> getServicesBlockedOnDestination(PerunSession sess, int destinationId) throws PrivilegeException, DestinationNotExistsException, FacilityNotExistsException {
+		Destination destination = getDestinationById(sess, destinationId);
+		List<Facility> destinationFacilities = perunBl.getFacilitiesManager().getFacilitiesByDestination(sess, destination.getDestination());
+
+		boolean isFacilityAdminAtLeastOnce = false;
+		for (Facility facility : destinationFacilities) {
+			if (AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN, facility)) {
+				isFacilityAdminAtLeastOnce = true;
+			}
+		}
+		if (!isFacilityAdminAtLeastOnce && !AuthzResolver.isAuthorized(sess, Role.PERUNOBSERVER)) {
+			throw new PrivilegeException(sess, "getServicesBlockedOnDestination");
+		}
+		return getServicesManagerBl().getServicesBlockedOnDestination(sess, destinationId);
+	}
+
+	@Override
+	public boolean isServiceBlockedOnFacility(PerunSession sess, Service service, Facility facility) throws PrivilegeException {
+		// Authorization
+		if (!AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN, facility) &&
+				!AuthzResolver.isAuthorized(sess, Role.PERUNOBSERVER)) {
+			throw new PrivilegeException(sess, "isServiceBlockedOnFacility");
+		}
+		return getServicesManagerBl().isServiceBlockedOnFacility(service, facility);
+	}
+
+	@Override
+	public boolean isServiceBlockedOnDestination(PerunSession sess, Service service, int destinationId) throws PrivilegeException, DestinationNotExistsException, FacilityNotExistsException {
+		Destination destination = getDestinationById(sess, destinationId);
+		List<Facility> destinationFacilities = perunBl.getFacilitiesManager().getFacilitiesByDestination(sess, destination.getDestination());
+
+		boolean isFacilityAdminAtLeastOnce = false;
+		for (Facility facility : destinationFacilities) {
+			if (AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN, facility)) {
+				isFacilityAdminAtLeastOnce = true;
+			}
+		}
+		if (!isFacilityAdminAtLeastOnce && !AuthzResolver.isAuthorized(sess, Role.PERUNOBSERVER)) {
+			throw new PrivilegeException(sess, "isServiceBlockedOnDestination");
+		}
+
+		return getServicesManagerBl().isServiceBlockedOnDestination(service, destinationId);
+	}
+
+	@Override
+	public void unblockAllServicesOnFacility(PerunSession sess, Facility facility) throws PrivilegeException {
+		// Authorization
+		if (!AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN, facility)) {
+			throw new PrivilegeException(sess, "unblockAllServicesOnFacility");
+		}
+		getServicesManagerBl().unblockAllServicesOnFacility(sess, facility);
+	}
+
+	@Override
+	public void unblockAllServicesOnDestination(PerunSession sess, String destinationName) throws FacilityNotExistsException, PrivilegeException {
+		List<Destination> destinations = ((PerunBlImpl) sess.getPerun()).getServicesManagerBl().getDestinations(sess);
```

Comment:
just use ``getServicesManagerBl()`` instead of ``((PerunBlImpl) sess.getPerun()).getServicesManagerBl()``

Reply:
This is not resolved. While it works, I would prefer using shorter version - to get bl layer manager from within entry manager.